rviz_rendering/src/rviz_rendering/ogre_render_window_impl.cpp

```diff
@@ -123,7 +122,8 @@ RenderWindowImpl::render()
   // Theoretically you can have one function that does this check but from my
   // experience it seems better to keep things separate and keep the render
   // function as simple as possible.
-  Ogre::WindowEventUtilities::messagePump();
+  // TODO(Martin-Idel): This should be removed
```

Comment:
Should this be resolved before merging? Why is the messagePump no longer required?

Reply:
Of course, I just forgot about it. It was moved to `OgreBites`, which I disabled because it needs SDL to build completely and throws a CMake warning if you don't. Also, this would be the one and only place where we need `OgreBites` so I tried to just disable it and never found anything wrong with it.

To be honest, I don't know what the message pump is actually supposed to do. If you look at the implementation (https://github.com/OGRECave/ogre/blob/f4709042cf7722757d86fa830fedea0985017d53/Components/Bites/src/OgreWindowEventUtilities.cpp) you'll see:
- it does nothing on Mac
- It does something on Linux, but it iterates over render windows which have to be registered. Since the corresponding function was never called, I don't think it does anything on Linux (and I couldn't see any change in functionality).
- It might do something on Windows (three lines that I'm inclined to copy if they are actually needed), but then again I don't see any difference.

Reply:
The Windows part seems relevant, as it seems to be processing GUI "MSG's" or something. I have no idea if we're using those, or if Qt is pumping them for us or not. Similarly the Linux code seems to be processing messages from the OS about change of focus and occlusion.

This line seems to indicate that on Windows it avoids the "not responding" behavior (happens when a program freezes):

https://github.com/OGRECave/ogre/blob/1fdf49ce312ee8f1c4426f826326601aaef1b521/Components/Bites/src/OgreApplicationContextBase.cpp#L550

Which makes sense that an application needs to do something for Windows to know whether it is or isn't responding. But again, I'd expect Qt does this for us?

---

I spoke with @iche033 about it and I think this function might be doing some useful things like disabling rendering of the render window when the window is occluded:

https://github.com/OGRECave/ogre/blob/f4709042cf7722757d86fa830fedea0985017d53/Components/Bites/src/OgreWindowEventUtilities.cpp#L335-L353

But I think it's only meant to be used by Ogre applications (demos and samples), and instead we could/should use the Qt events in a similar fashion.

I'm not sure we need to do that now, but perhaps remove this and make an issue pointing out that our behavior may have changed since we no longer call this function, and that we should consider emulating some of its behavior with Qt events instead, as follow up work.

---

Also, there's this issue: https://github.com/ros2/rviz/issues/328

But I don't think its directly related. I'm not sure if it is speaking about losing focus within a Qt widget or like losing focus of a window or something.

Reply:
#328 is definitely a Qt-Problem (the focus of the display widget is lost), so that's not related.

I did a bit more digging and also found https://forums.ogre3d.org/viewtopic.php?t=66358 where it says 
>If you are using Qt, it also pumps messages indiscriminately of window. You shouldn't need to call WindowEventUtilities::messagePump() unless your Qt window and Ogre window are on separate threads

which is in line with what you are saying and is not contradicted by other OGRE devs in the same thread [but is contradicted by the official wiki at https://wiki.ogre3d.org/Integrating+Ogre+into+QT5 which I guess was the original inspiration].

For now, I'll leave a comment and will create an issue later.

Reply:
Sounds good to me! Thanks for looking into it.